src/main/java/com/michelin/cio/hudson/plugins/rolestrategy/RoleBasedAuthorizationStrategy.java

```
@@ -233,10 +233,147 @@ private void addRole(String type, Role role) {
    */
   private void assignRole(String type, Role role, String sid) {
     RoleMap roleMap = this.grantedRoles.get(type);
-    if(roleMap != null && roleMap.hasRole(role)) {
+    if (roleMap != null && roleMap.hasRole(role)) {
       roleMap.assignRole(role, sid);
     }
   }
+  
+  /**
+   * API method to add roles
+   * @param request has to contain following parameters 
+   *    type (globalRoles, projectRoles)
+   *    roleName
+   *    permissionsIds
+   *    overwrite (true, false)
+   * example: curl -X POST localhost:8080/role-strategy/strategy/addRole --data "type=globalRoles&roleName=ADM&permissionIds=hudson.model.Item.Discover,hudson.model.Item.ExtendedRead&overwrite=true"
+   * @param response
+   * @throws IOException
+   */
+  public void doAddRole(StaplerRequest request, StaplerResponse response) throws IOException {
+    Jenkins.getInstance().checkPermission(Jenkins.ADMINISTER);
+    String type = request.getParameter("type");
+    String roleName = request.getParameter("roleName");
+    String permissions = request.getParameter("permissionIds");
+    String overwrite = request.getParameter("overwrite");
+    
+    if (type != null && roleName != null && permissions != null) {
```

Otherwise you should throw an error